design_drafts/components_architecture_and_urdf_examples.md

```diff
@@ -0,0 +1,424 @@
+# ROS2 Control Components Architecture and URDF-Description Examples
+
+## Architecture
+
+![ros2_control Components Achitecture][ros2_control_arch_resource_manager]
+
+[ros2_control_arch_resource_manager]: images/components_architecture.png "ROS2 Control - Components Architecture"
+
+## URDF Examples
+
+ros2_control implementation examples are presented for the following robot/robot-cell architectures:
+
+1. Industrial Robots with only one interface
+2. Industrial Robots with multiple interfaces (can not be written at the same time)
+3. Robots with multiple interfaces used at the same time
+4. Industrial Robots with a sensor integrated into the robot's control box
+5. Industrial Robots with a sensor connected to ROS computer
+6. Modular Robots with separate communication to each actuator
+7. Modular Robots with actuators not providing states and additional sensors (simple Transmissions)
+8. Modular Robots with separate communication to each "actuator" with multi joints (Transmission Example) - (system component is used)
+9. Sensor only
+10. Actuator Only
+
+
+Note:
+  * Everything within the `<classType>` tag is implemented as a plugin.
+  * The examples below have some `<param>` tags defined for each plugin which are primarily for demonstration, not part of a pre-defined xml schema. Components can define their own parameters.
+#### 1. Industrial Robots with only one interface
+  * the communication is done using proprietary API to communicate with robot control box
+  * Data for all joints is exchanged in batch (at once)
+  * Examples: KUKA RSI
+
+```xml
+  <ros2_control name="2DOF_System_Robot_Position_Only" type="system">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_Position_Only</classType>
+      <param name="example_param_write_for_sec">2</param>
+      <param name="example_param_read_for_sec">2</param>
+    </hardware>
+    <joint name="joint1">
```

Comment:
> this joint is purely virtual and abstract to any hardware specific implementation

I do not agree. On this level, in the `RessourceManager` all joints are actually the real joints of the hardware. First when we go into `ControllerManager` virtual joints are used. (This is also shown, not in full detail, in the image)

Reply:
Then I still don't understand what's happening with joints. In my understanding, the term joint here described something abstract which encapsulates common pattern like joint limits etc before these values are then finally passed to the system implementation.

Reply:
There's the forward facing XML format we are trying to set up and the code that this will eventually tie into. I would 100% consider joint_handle an implementation detail nobody should care about when defining/reading a URDF. No matter how this is implemented, the URDF should never refer to handles, let's keep it at the abstract terms level.